internal/incident/incidents.go

```diff
@@ -31,17 +94,9 @@ func GetCurrent(
 
 	if currentIncident == nil {
 		ir := &IncidentRow{}
-		incident := &Incident{
-			Object:           obj,
-			db:               db,
-			logger:           logger.With(zap.String("object", obj.DisplayName())),
-			runtimeConfig:    runtimeConfig,
-			configFile:       configFile,
-			Recipients:       map[recipient.Key]*RecipientState{},
-			EscalationState:  map[escalationID]*EscalationState{},
-			SeverityBySource: map[int64]event.Severity{},
-			Rules:            map[ruleID]struct{}{},
-		}
+		incident := NewIncident(db, runtimeConfig, configFile)
+		incident.Object = obj
+		incident.logger = logger.With(zap.String("object", obj.DisplayName()))
```

Comment:
`object` and `logger` should be arguments of `NewIncident()` as well. After all, if they aren't set, that object won't function properly.